### PR TITLE
Redesign navbar

### PR DESCRIPTION
**Pull request by benmanley**
#### header.styl
- Change background-color of .header-wrap and .hero-stats
#### menu.jade

Redesign navbar to:
- Use more custom css and less Bootstrap
- Add Task and Options menu items
- Don’t hide Task and Options in mobile view
- Toggle whole link menu rather than using dropdown
- Improve mobile layout
#### menu.styl
- Create new stylus file for custom navbar styles
- New toolbar colour scheme
- Layout changes
- Improve look of collapsed toolbar chevron (.collapse-toolbar)
#### index.styl
- Import menu.styl
- Move .collapse-toolbar css to menu.styl
#### npcs.styl
- Adjust .toolbar-bailey-container
- Adjust position of .npc_bailey_head

UUID:1322727e-9911-4ad4-aea9-a5ebd8cba826
User: Pixel

### CODE REVIEW

File: public/css/menu.styl [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Custom styles for the menu aka toolbar aka navbar
+.navbar
+  color: #222
+  background-color: #f8f8f8
```
Comment (deilann) [inline public/css/menu.styl]: I thought no one liked white, either? Thought grey was the solution here?

**Comment by benmanley** [comment]
@deilann Yes. Looks like it should have been #f5f5f5. I think that was just my mistake. #f8f8f8 is a light grey, so I didn't notice it. I've updated it. Thanks.

File: views/shared/header/menu.jade [hunk 2]
```diff
@@ -1,128 +1,122 @@
...
-        span.glyphicon.glyphicon-chevron-up.white
-
-  .navbar-header
-    button.navbar-toggle(type='button', ng-click='isNavbarCollapsed = !isNavbarCollapsed', ng-init='isNavbarCollapsed = true')
-      span.sr-only Toggle navigation
-      span.icon-bar
-      span.icon-bar
-      span.icon-bar
+      a(ui-sref='tasks')
+        span Tasks
```
Comment (paglias) [inline views/shared/header/menu.jade]: env.t('tasks')

File: views/shared/header/menu.jade [hunk 3]
```diff
@@ -1,128 +1,122 @@
...
+    li
+      a(ui-sref='options.profile.avatar')
+        span Options
```
Comment (paglias) [inline views/shared/header/menu.jade]: env.t('options')

**Comment by paglias** [comment]
@benmanley Ok, I've checked out the PR and are testing it now.
 I can't push to your branch so I'm just making comments.
1. The sync button has an `important` css rule that make it black

selezione_044
1. The `hero-stats` class has a different color from the rest of the header `c6c6c6` vs `c0c0c0`

selezione_045

**More general comments**
1. The button to expand the toolbar color, white, doesn't look so good on the header
2. We can remove the tabs now? Although I have to say that I don't find very convenient having to scroll to the top of the page to visit another part of the site. It would be cool to have the toolbar showing above the header even if you scrolled down the page and click to expand it, or at least have the page scroll to the top to show it.
3. It doesn't change the fact that for example in the market you don't see the amount of gold, gems and silver you have

**Comment by benmanley** [comment]
Thanks @paglias.
1. I made the Sync button black to indicate somehow that it had a different function to the rest of the menu. Needed more work. I’ve moved it out of the submenu entirely and into the top navbar.
2. The colour difference is deliberate. I wanted to give .hero-stats a different colour to the background for your party. I’ve set up a base colour variable for the header to make it easier to tweak the colour scheme, so maybe we can see what people think of the new colour scheme.

**General comments**
1. I’ve changed the colour back to the default
2. No. The tabs need to stay until I’ve done more work to redesign that bit of the navigation. You’re right that it won’t be convenient until that’s done. I’ve made the toolbar fixed for desktops and static for mobile. Should be an improvement.
3. I think the toolbar needs to be non-collapsible so that you can always see gems/gold/silver. I’ve tried removing the option to toggle it.

One thing I couldn’t work out was how to make the `.toolbar-submenu` hidden by default. (It's obscuring the header when you first load the page until you minimize it.)

**Comment by deilann** [comment]
Can you post a new screenshot?

**Comment by benmanley** [comment]
#### Submenu closed

screen shot 2014-03-17 at 00 06 57

#### Submenu open

screen shot 2014-03-17 at 00 07 43

#### Mobile submenu closed

screen shot 2014-03-17 at 00 11 26

#### Mobile submenu open

screen shot 2014-03-17 at 00 11 41

Mobile menu will neaten up if we can get currency smaller and amalgamate notifs. I reckon it's still an improvement, but see what you think.

**Comment by deilann** [comment]
I feel like the toolbar is still too light. It's not distinguishable from
the header for me.

But that's very much a YMMV.

**Comment by lefnire** [comment]
@benmanley I'm digging this. The only thing I'm not too keen on is the 2-layer menu system. Dunno what it is - maybe because you have to close it up when you're done (more clicks); very likely it's just personal preferences. I'm pushing it up to [beta](https://beta.habitrpg.com) so people can start playing with it & provide feedback. Don't be afraid to force-push changes, i'm working off a temp branch (dunno if you're force-pushing, or whatever's your workflow). Let me know as you want to push new changes to beta and I'll get them up.

Incidentally what was it in the current menu system that was snagging you? Was it that the list was too long for responsive toolbar? Also, should we nix [this puppy](https://www.evernote.com/shard/s17/sh/24ec1482-0c65-4049-a7c4-90e8a9debb7f/a317b8fd1a618f8fc4c59a7549f940cd) now?

**Comment by deilann** [comment]
Biggest issue so far-- If I open the sub-menu, then click on a link, it
should collapse; I should not have to close it.

**Comment by deilann** [comment]
Whenever I refresh, the sub-menu reopens.

**Comment by benmanley** [comment]
@lefnire I'm using the Github app as I am scared of breaking the internet. I'm not sure exactly what it does when it syncs. I can push via the terminal but I'm not that familiar with the commands. 
My last two commits could be pushed if you can.

@deilann I think I just fixed the issue with the submenu reopening.

Agree with both of you about closing the menu when clicking a link, but not sure how to achieve it at the moment. Will look at your other comments tomorrow. _z z z_

**Comment by lefnire** [comment]
pushing

**Comment by paglias** [comment]
@benmanley now I totally love it and having the toolbar sticky I think the tab navigation is even more useless

**Comment by benmanley** [comment]
@lefnire
Push to beta:
- Close submenu when a navigation link is clicked
- Remove gold and silver totals from Reward list 

The puppy is nixed. Let me know if that all looks OK.

Keeping the submenu in the top toolbar caused a couple of issues:
1. Until we amalgamate notifs, that area of the toolbar can get rather long. If the browser was sufficiently minimised, the submenu forced the notifs, etc to a second row. Didn’t look that good.
2. In mobile view, it meant that Bailey, notifs, currency, etc fell below the submenu. I think it’s probably better for them to be above.

Also, now that the toolbar disappears again when you click a link, you’re not actually clicking more than you were, even though you can’t see the icons initially.

Thanks @paglias. For now, I think we need to keep the tab navigation just for easier navigating within sections, e.g. from Pets to Mounts.

**Comment by benmanley** [comment]
@deilann 
@Mihakuu is also unsure about the toolbar colour.
I'll take a stab at improving it, but a lot of the CSS is still based on bootstrap, so the current background-color options need to be either close to black or close to white.  I'll need to overhaul the CSS to make more drastic changes.

**Comment by paglias** [comment]
@benmanley from a design point of view there's an area of some pixels outside the menu items in the toolbar. You can see it under a menu item when you hover it.
selezione_046

If we don't want to remove at all the tabs at least we could remove the main navigation system and keep only the sub-section one.

**Comment by deilann** [comment]
Close to black, but not actually black would be preferable in my eyes.

**Comment by lefnire** [comment]
pushed. really coming together, @benmanley - thanks so much for all your work on this! Anyone object to this going live?

**Comment by Mihakuu** [comment]
I agree with @deilann about the color

**Comment by benmanley** [comment]
Thanks @lefnire
Push to beta:
- Fix gap at bottom of toolbar and remove unused code

I've fixed the unwanted space @paglias mentioned above.

Since you're happy with the changes, I've removed some code that I'd commented out or had become redundant.

**Comment by brightbold** [comment]
So excited about this! The sticky toolbar is awesome, and I love having access to my gold count in the market.

When the notifications get amalgamated (yes please!), would we be able to put the top level nav back in the main toolbar area? Like @lefnire I don't love having to expand and collapse the menu options. In my ieadl world, I'd like to see Tasks User Social Inventory Settings Help all across the top, and have the drop-downs appear on hover (on devices that support hover) rather than click. That would allow you to get from one place to another in a single click! 

Will you need to bring back the collapsible toolbar option for mobile? I haven't tried it yet (it's late and my phone is all the way across the room and I'm too lazy to go get it, sorry) but it seems like people might want it on small screens.

**Comment by benmanley** [comment]
@lefnire Don't go live until you’ve pushed my latest commit. I broke the logout link. Should be fixed now.

**Comment by benmanley** [comment]
Thanks @brightbold. Yes, I definitely think we should revisit once everything’s been shrunk as much as possible. Personally, I’m not a great fan of dropdowns on hover as they can be fiddly. But if, as you suggest, we have words rather than icons, at least the hover target would be that much bigger.

I don’t think you need the toolbar to be collapsible on mobile if it isn’t sticky (which it isn't right now). I reckon mobile navigation needs a rethink anyway.

**Comment by SabreCat** [comment]
I'm actually not digging the auto-close when clicking a link. Often times I want to hop into Tavern, then my Guilds, then my Party, and so forth. It's actually making that flow more click-heavy if I try to use the new navbar, such that my path of least resistance leads me back to using the tabs.

EDIT: It would make sense for it to flip back to the avatar+party view when I go back to Tasks.

**Comment by deilann** [comment]
But at the same time, always having to close it is super annoying. The best
solution here would be having it be hover and having a way to make it stick
(clicking)?

**Comment by SabreCat** [comment]
Hover+pin sounds viable. Having a way to stick the navbar would be nice for distraction management too, no sprites and colored bars to draw the eye ^.^

**Comment by benmanley** [comment]
@SabreCat @deilann Once we've squished down all the notifs and so on, you could have more top level navigation, say:

Tasks – User – Party – Tavern – Stable – Market – Guilds

Or 

Tasks – User – Party – Town

with a hover over "Town" to give "Tavern – Stable – Market – Guilds"

**Comment by benmanley** [comment]
On a related note – how often do people use the sync button? Does it need to be top level, or could it be a subsection of settings?

I'm wondering about a cog icon (or similar) on the far right for Settings. Could Sync (and also Help) be a subsection of that?

My idea is essentially – navigation on the left; notifications, info & settings on the right.

Anyway, it's getting late where I am. _z z z_

**Comment by deilann** [comment]
Sync needs to be easy to find and on the top level, please. It makes
support a lot easier.

**Comment by SabreCat** [comment]
Sync within single click reach is crucial, yeah. I do a lot of bebopping between computers, and hitting a Sync gets me up to date and oriented when I arrive at my new context.

**Comment by benmanley** [comment]
OK. Thanks. I'll leave sync where it is.

**Comment by wc8** [comment]
+1 for more top level navigation and pin to hide the header

If 'Tasks – User – Party – Town, with a hover over "Town" for "Tavern – Guilds – Stable – Market"
is chosen, it might make sense to have a hover over "User" for "Achievements — Appearance — Equipment — Profile — Settings" etc.

**Comment by lefnire** [comment]
Alright @benmanley, merge time. Will be pushing to prod tomorrow. Any new changes on a new PR please :)

**Comment by lefnire** [comment]
Forgot your UUID, what'sit again? contrib.lvl+=2

**Comment by deilann** [comment]
I've got it @lefnire `1322727e-9911-4ad4-aea9-a5ebd8cba826` I'll just go ahead and lvl+=2 while i have it up.

**Comment by deilann** [comment]
✓

**Comment by deilann** [comment]
...not sure why I have scarily good memory for Github to Habit username knowledge, but can't ever remember if I ate in the last 8 hours...

**Comment by lefnire** [comment]
Thanks ryan!

**Comment by deilann** [comment]
np

i know you're super busy, so jsyk, if it makes your day easier and if you
remark to give contrib levels, I can assign them, saving you time

unless of course, you find joy in assigning contrib levels, in which case,
feel free to ignore this offer. ;)

**Comment by lefnire** [comment]
You're amazing! Will do

**Comment by deilann** [comment]
o7

**Comment by benmanley** [comment]
Thanks all.

**Comment by annesoares** [comment]
Hi! So, basically, I logged into Habit today and that's what it looks like for me. I can't really see my tags. For the tabs at Options and other pages, it shows at the same place. My screen resolution is 1366 x 768 and I'm using Chrome 33.0.1750.154, if that helps. I tried to zoom out, but it still looks the same.
image

And I personally prefer the old toolbar, with all the icons and collapsible. But I loved the idea of seeing coins and gems at the toolbar =)

---

EDIT: today it's working fine =D

**Comment by AranelS** [comment]
I have the same resolution (it's the optimal resolution for my widescreen laptop) but am using Firefox.  I can see my tags, but only about three dailies on the first screen.  The Tavern screen, which I got a shot of, is almost entirely "below the fold". I got it to be more functional by un-sticky-ing the header, but of course I still have to scroll past it the first time I load the screen.  

I would really prefer either a much smaller menu bar (this one is fixed-height with a lot of useless open space around it) or an option to make the menu bar go away. 
ss

**Comment by wc8** [comment]
I'm looking forward to the notification consolidation I wonder if the existing images could be used as placeholders as @benmanley did for this [image](https://f.cloud.github.com/assets/3317594/2334261/a995c154-a476-11e3-973c-9ac3703737b9.png), at least until pixel art replacements are designed if needed?

BTW, someone suggested in the Tavern that the navigation area should scroll back to the top when switching tabs/links.

**Comment by rkuykendall** [comment]
I just wanted to chime in that I love the designs that were recently pushed and the upcoming work look great too. I love that you're tackling the awful double-tabs.

My only input is that the 200px position:fixed top section makes the site feel really cramped and cluttered, since I'm losing 30% of my window height to navigation. Since Habit pages really aren't very long, is this necessary? I never find myself thinking "Man, I wish I didn't have to scroll all the way to the top to get back to the navigation or see my gold."

Keep up the good work, and I hope to someday contribute to this awesome project!

**Comment by brightbold** [comment]
> I never find myself thinking "Man, I wish I didn't have to scroll all the way to the top to get back to the navigation or see my gold."

Actually, @rkuykendall, I used to think this all the time! So I love the fixed position toolbar. But if you don't want your avatar and stats to be taking up space when you scroll down (which I also don't), go to your setting page and uncheck "sticky header." Then the toolbar stays fixed but the 147px avatar/stat area scrolls. You may find that you don't mind (or even appreciate) the ever-present toolbar once you're only losing 53px of screen real estate!

**Comment by deilann** [comment]
I'm with BrightBold here. I think that _all_ the time. :P

I don't use a sticky header, to liberate screen space. But, I would like a
setting that allows you to unsticky the toolbar as well; separate from the
header.

**Comment by rkuykendall** [comment]
Oh, I should have checked the settings first! Sorry for clogging up design conversation with that turned out to be a support issue. The stick toolbar is not a concern of mine, though I might disable it if there was an option. I'm really glad yall are putting so much thought into the UI design. Keep it up!

**Comment by deilann** [comment]
Don't worry about it. :)

On Sun, Mar 23, 2014 at 4:19 PM, Robert Kuykendall <notifications@github.com

**Comment by benmanley** [comment]
The toolbar could certainly be a few pixels narrower. To do it, I think I need to rewrite the toolbar CSS without using bootstrap. I'm also working on reducing the amount of tabbed navigation which will free up more space too.

@AranelS Sounds like your best bet would be to switch off the header entirely in settings.

**Comment by AranelS** [comment]
I would rather turn off the toolbar and keep the header. Otherwise what is the point of getting my little character all dressed up, with a pet and a mount that match?  ;)

**Comment by Mihakuu** [comment]
one thing I would like to see back is the chat notification for a certain page get marked read if you load in on that page. I typically join habit on a guild page, using the mobile to check off tasks and look at party chat, but now whenever I join that page I still have the chat notif until I leave the page and come back which is a tiny but annoying.

**Comment by deilann** [comment]
@Mihakuu That never worked for me. :c I don't think it was ever implemented.

**Comment by Mihakuu** [comment]
It used to work when the bar was black. I never had an issue with it before it changed.

**Comment by deilann** [comment]
It never worked for me when teh bar was black and the redesign doesn't
change the function of the notifs, so I'm not sure why it would change it.
Hmmm.

**Comment by larouxn** [comment]
I believe the menu arrow at the top should naturally fave right then down upon press (expanding the menu).

Just a small improvement that I can write up and submit in a pull request.

It just doesn't make logical UX sense for the arrow to face down and upon the menu appearing below, it swing to facing up. I guess it is supposed to represent the action of clicking to make menu come down then up, but I believe it makes more sense to face right then down.

Mockup:
right
down

Opinions?

**Comment by MaxHedrm** [comment]
That does make more sense, and is a more common implementation. I thought that menu was a bit odd. :)

**Comment by benmanley** [comment]
@larouxn @MaxHedrm 

Hi guys. I would say that the right facing arrow is more common in vertical menus, where the arrow is placed to the left of the text, e.g. Wikipedia

screen shot 2014-03-26 at 20 43 33

In a horizontal list, a downwards pointing arrow on the right of the text is not unheard of ;-)

screen shot 2014-03-26 at 20 40 24

screen shot 2014-03-26 at 20 41 25

screen shot 2014-03-26 at 20 48 46

**Comment by MaxHedrm** [comment]
True, however, in both cases the arrow is next to a word, not on its own. And, for the Amazon & Google implementations the arrow doesn't flip to point up. When movement is used I think the implementation where it starts out pointing right is cleaner and more intuitive.

**Comment by benmanley** [comment]
I agree about the spacing. I'm about to do a PR and I'll add it to that. Plus, I think the menu is going to end up looking more like:

Item 1&nbsp;&nbsp;&nbsp;Item 2 ∨&nbsp;&nbsp;&nbsp;Item 3&nbsp;&nbsp;&nbsp;Item 4 ∨

so, I think that will solve things to some extent.

Personally, I don't understand the logic of the arrow pointing right and away from its associated text. To me, it doesn't signify anything, whereas, down means down and up means up..

**Comment by lefnire** [comment]
I agree with @benmanley. I've always understood from interfaces: down-arrow means drop-down, right-arrow means expand (accordion)

**Comment by MaxHedrm** [comment]
Maybe I'm missing something here, but there is no text associated with the arrow. It has an entirely separate function from Options.

My biggest problem with the direction of the arrow now is probably the animation, I'm not sure it adds any value.

**Comment by larouxn** [comment]
So is the consensus that @benmanley will change it so by default the arrow is facing right then when clicked to show the menu will animate clockwise to a down position?

**Comment by MaxHedrm** [comment]
It looks like he's leaving it pointing down, not sure it if will still rotate... https://github.com/HabitRPG/habitrpg/pull/3155

**Comment by benmanley** [comment]
@MaxHedrm 
Currently, the arrow is out on its own, but the submenu it triggers is essentially a list of options. Clicking Options happens to default to Profile > Avatar, but I'm planning an Options landing page instead which is yet to be done. So I thought it made sense to associate the arrow with Options by moving it nearer.

In general, the toolbar is work in progress and I think it will all seem more sensible when it's done. Hopefully.

I like the animation. I’d welcome more opinions before taking a decision to remove it.

**Comment by MaxHedrm** [comment]
It has a different function, so it is NOT related. If they are too close together, the assumption will be that they have the same function when clicked and they do not. They have a very different function.

As for the animation, do you like it because it is nifty, or because it actually serves a purpose? If it's just because it's nifty, leave it out.

**Comment by larouxn** [comment]
@benmanley I'm totally fine with the animation. I'm just proposing it start facing left, animate down on click, and then back up to right when clicked again. (animates towards the right side)

**Comment by lefnire** [comment]
Another possibility: menu Been seeing that more & more these days ([eg](https://codeclimate.com/)), then we could skip the animations / direction issues. Just one option.

Ok all *clears throat*. I'm giving @benmanley creative license here. Certain UX bits are right/wrong (as convention dictates), and it's good to point out when implemented incorrectly by mistake. Others (most of this thread) are opinion. Obviously, conversation / debate is good, healthy, and inevitable. But when it comes decision time, we either (a) do a poll, for the bigger more controversial things; (b) or let the "doer" (@benmanley) decide. For something like a triangle icon, let's just let Ben do his thing. I know this sounds heavy-handed, but we've lost a primary contributor in the past due to Github bullying, and I want to be careful that doesn't happen again.

**Comment by deilann** [comment]
I like the three lines!

But I agree. It should be up to @benmanley

**Comment by MaxHedrm** [comment]
I'm sorry if it came across as bullying, it really wasn't intended to be. I'm just a bit hypersensitive to UI/UX issues. :smile:

The 3 lines thingy (I remember seeing a name for them but I can't remember it right now) is a popular option, especially in the mobile app world.

**Comment by wc8** [comment]
@lefnire :+1: for doers deciding details, provided the founder's ok with it. 
Feedback's good, but if something's not going to hamper user's playing style/productivity, makes sense the designer/builder should decide.

**Comment by benmanley** [comment]
Thanks all.

@MaxHedrm Don’t sweat it. I take most of my UX/UI inspiration from [Webkingusa](http://webkingusa.com), but I’m open to other points of view. The navigation redesign is half finished and things are evolving as I go, so I think discussions on the details are somewhat moot anyway for now.

@lefnire Three lines could work. I’m wrestling with it all at the moment. Trying to make it more mobile friendly too.

**Comment by wc8** [comment]
Happened to see Google's toolbar mobile on Firefox today. I'm not sure if it'd work with Habit's menus, but it certainly seemed nice and neat. The arrangement was something like so:

menu  [ ]() [Web](link) [Images](link)

**Comment by wc8** [comment]
@benmanley would it be possible to have a section which displays a list of one's guilds for easier navigation?

**Comment by deilann** [comment]
I would love that as well. Especially if it could be combined with a
selective notifs feature. So I could set a guild as "starred" and have only
"starred" guilds show up on the list.

**Comment by brightbold** [comment]
Yeah if we're sticking with having the full menu expand when something is clicked on wide screens (although my preference is still for individual menu item dropdowns once we get notifications collapsed and have space), there's an advantage in using the three-line menu icon, since that's common on mobile and then we'll already have it in place and have people used to clicking it.

But whatever you decide @benmanley, this is looking great and is a great improvement over all those tabs. I can't wait until we can get rid of the redundant tabs!

**Comment by benmanley** [comment]
I'm moving towards individual dropdowns for desktop. Seems more popular. 

As a straw poll, how many guilds are you guys in?

**Comment by larouxn** [comment]
Zero at this point, though when I do join some, I'll probably join about 2-3.

**Comment by brightbold** [comment]
Fourteen. I'd say that breaks down into only about 3-4 guilds where I want to be notified of every new post, 6 that I like to keep my eye on and would love mentions notifications if we ever make that possible, and the rest that I occasionally check but could drop (and would if I didn't know that notifications improvements were coming!)

**Comment by brightbold** [comment]
Pro Tip: If you make the first sentence of your comment the number of guilds you're in, followed by a period, Github thinks you're creating an ordered list and changes it to "1." So don't do that. _Not that I did that or anything. Just, you know, trying to help out others._

**Comment by wc8** [comment]
6 or so. I trimmed down because of too many notifications...
BTW, individual dropdowns for desktop definitely my preference. Not sure on mobile
2¢

**Comment by benmanley** [comment]
OK, thanks all.

BTW, in view of the completely warranted changes to HabitRPG, the new menu icon will now be

:tomato:

**Comment by larouxn** [comment]
Perfect. I'm all for tomato options.

**Comment by casicastiel** [comment]
...... Twenty. :P At least .... 8-10 of those I want to know of every new post in, I think.

**Comment by deilann** [comment]
31 guilds at the moment.

**Comment by wc8** [comment]
@benmanley I'm finding I frequently confuse gem amount for gold amount. I wonder if there's a way to make the connection to the icon clearer.

**Comment by benmanley** [comment]
@wc8 Yes. The plan is to use the coins / gems as a background to each amount, but I need some new pixel art before I can do that. In the meantime, I've matched the colour of the text to the images to make it a bit clearer. Will commit it to PR #3155.

**Comment by deilann** [comment]
How will that work if someone has 26,000 gold?

**Comment by MaxHedrm** [comment]
I would say use icons that are smaller than the current gem, but a smidge taller than the current coins (why aren't they round?). Then just start the number on the icon. So maybe the first number and a half or so is on the icon.

**Comment by deilann** [comment]
The current coins aren't round because you're looking at them from an
angle, from what I can see.

**Comment by MaxHedrm** [comment]
But the gem isn't at that angle, so it makes it a bit wonky.

**Comment by deilann** [comment]
Agreed -- but before, the gems and the gold were never shown together.

**Comment by wc8** [comment]
I wondered that too, but I suppose the amount could be centered on the gem/coin, perhaps a little blue/yellow/silver background behind any overhanging text. I suppose it's something for the pixel artists @lemoness and @benmanley to sort out.
Speaking of which, @lemoness, has the need for pixel art been noted?

**Comment by benmanley** [comment]
I imagined rounding amounts over 999 [something like this](https://camo.githubusercontent.com/702794e6cba1f0cac7433f21c3814782785390a4/68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f333331373539342f323333343236312f61393935633135342d613437362d313165332d393733632d3961633337303337333762392e706e67).

@lemoness put [a pixel art request](https://trello.com/c/Xey7Guzc/28-miscellaneous) on Trello, but I think I need to explain what I need in a bit more detail. As you say, the current icons don’t really match as they haven’t been put together until now. I was hoping for something flat and circular, more along the lines of the achievement medals. I’ll get onto it once I’ve finished my latest PR.

**Comment by MaxHedrm** [comment]
I'll try to mock something up later. I better put a todo in for it. ;)

**Comment by lemoness** [comment]
Yeah, the simplest thing would be for @BenManley to comment with the
specifics of his needs :) Or @MaxHedrm can just do it!

**Comment by benmanley** [comment]
Yes. @MaxHedrm, that would be great. I'd say they need to be about 30px in diameter and plain in the middle to ensure the numbers are legible.

**Comment by MaxHedrm** [comment]
Here's what I was thinking. I cheated & just scaled the existing coins for the first pass. I also made the font size larger. And made the menu bar & subscribe button thinner, since people had requested that.

I think the real solution would be to keep the font size the same & go smaller with the gem & coins. Perhaps 24px? That would let the menu bar be even a bit thinner. I'll try a mockup of that next.

hrpg-menubar-30px

**Comment by benmanley** [comment]
Thanks @MaxHedrm. You might want to check out #3155 before you put too much work into the toolbar as a whole. That PR is nearly ready to go and includes a narrower toolbar. Also, bear in mind that Bailey appears in the toolbar, so there’s a limit to how skinny it can be.

We’ve also added some code to display gold > 999 in the format “1.0K” so that should contain the numbers sufficiently to put a background behind them.

For me, the overlapping text in your mockup feels slightly unbalanced. I would say the image should either be to the left of the text or behind it.

This is how the toolbar is looking in #3155:

new-toolbar

I’ve dropped in a mockup of currency with images as backgrounds to show how that might look, but they would need to be worked up into proper pixel art.

**Comment by deilann** [comment]
Looking at that mockup, it would seem kind of natural to put a "collapse
toolbar" in the settings list, then keep the gear in the upper right corner
so they can then "uncollapse"

**Comment by deilann** [comment]
That way, no more buttons.

**Comment by annesoares** [comment]
I really liked this design, @benmanley . One question: will there be a way to see exactly how many gold I have? Like, if I have 1.678 gold. And gems too XD

**Comment by brightbold** [comment]
I was thinking that if we need to truncate gold amounts (e.g. 1.6K) for design purposes, there should be somewhere in the interface you could see the exact number, either by hovering over the 1.6K or just on some screen somewhere (the way you can see gold amounts in another person's popup profile.)

**Comment by MaxHedrm** [comment]
Yeah, I knew that was going on, I just didn't know how much they were shrinking it. I just did that to show what I was thinking (easier than trying to describe with words :smile:).

I like the mouse over idea for the full amount. So, 1.6k would be 1652 or whatever in the mouseover. With the #.#k they probably can fit in the 30px icon pretty well. I'll steal your image, @benmanley, and use it as the sizing/testing for a better gem & coin. But right now it's food time. :wink:

**Comment by benmanley** [comment]
@annesoares @brightbold @MaxHedrm I've asked @paglias about adding the full gold amount to the popover. You'll also be able to check it in your stats.

**Comment by deilann** [comment]
I'd imagine it would also be available by rewards still.

**Comment by benmanley** [comment]
It's already deleted from Rewards. Better for it not to appear all over the place, I think. But it makes sense to me that you can see the full amount in your stats.

**Pull request merged**